dashboard/app/helpers/proxy_helper.rb

```diff
@@ -0,0 +1,67 @@
+require 'net/http'
+require 'uri'
+
+module ProxyHelper
+  def render_proxied_url(location, allowed_content_types:, expiry_time:, infer_content_type:, redirect_limit: 5)
+    if redirect_limit == 0
+      render_error_response 500, 'Redirect loop'
+      return
+    end
+
+    # Give up if the host doesn't respond within 3 seconds to avoid
+    # tying up Rails thread.
+    url = URI.parse(location)
+    raise URI::InvalidURIError.new if url.host.nil? || url.port.nil?
```

Comment:
I went to fix this, but it looks like we already catch this exception and return 400 later in this file.